vmm/src/device_manager.rs

```diff
@@ -450,6 +450,50 @@ impl DeviceRelocation for AddressManager {
                         .register_ioevent(event, &io_addr, NoDatamatch)
                         .map_err(|e| io::Error::from_raw_os_error(e.errno()))?;
                 }
+            } else {
+                let virtio_dev = virtio_pci_dev.virtio_device();
+                let mut virtio_dev_locked = virtio_dev.lock().unwrap();
+                if let Some(mut shm_regions) = virtio_dev_locked.get_shm_regions() {
+                    if shm_regions.addr.raw_value() == old_base {
+                        // Remove old region from KVM by passing a size of 0.
+                        let mut mem_region = kvm_bindings::kvm_userspace_memory_region {
+                            slot: shm_regions.mem_slot,
+                            guest_phys_addr: old_base,
+                            memory_size: 0,
+                            userspace_addr: shm_regions.host_addr,
+                            flags: 0,
+                        };
+
+                        // Safe because removing an existing guest region.
+                        unsafe {
```

Comment:
I guess the same thing needs to be in the removal case?

Reply:
Yes but this will be done in a follow up PR. The two cases are kind of disjoints since here it is specific to PCI BAR remapping. I mean to say it's only because the shared region is behind a BAR that we have to handle this use case.
There's a more generic use case for unplugging fs and pmem that needs to be taken care of to make sure the KVM userspace mapping goes away.

Reply:
Okay but please make sure you do that ASAP so we don't forget!

Reply:
Oh yes that's the next PR I'm working on :)